unstructured/ingest/connector/kafka.py

```diff
@@ -0,0 +1,314 @@
+    @BaseSingleIngestDoc.skip_if_file_exists
+    def get_file(self):
+        self._create_full_tmp_dir_path()
+
+        pdf_data = base64.b64decode(self.raw_content)
+
+        with open(self.filename, "wb") as file:
+            file.write(pdf_data)
+
+
+@dataclass
+class KafkaSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    """Source connector for Kafka.
+    Main job is to consume from a Kafka topic and create instances of
+    KakfaIngestDoc.
+    Note that messages have the format of:
+    <filename>: the name of the file (with correct file extension)
+    <content>: base64 encoded (whether was binary or not)
+    """
+
+    connector_config: SimpleKafkaConfig
+    _consumer: t.Optional["Consumer"] = None
+
+    def check_connection(self):
+        try:
+            self.kafka_consumer
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {e}")
+
+    def initialize(self):
+        topic = self.connector_config.topic
+        logger.info(f"Subscribing to topic: {topic}")
+        self.kafka_consumer.subscribe([topic])
+
+    @property
+    def kafka_consumer(self):
+        if self._consumer is None:
+            self._consumer = self.create_consumer()
+        return self._consumer
+
+    @requires_dependencies(["confluent_kafka"], extras="kafka")
+    def create_consumer(self) -> "Consumer":
+        from confluent_kafka import Consumer
+
+        is_confluent = self.connector_config.confluent
+        bootstrap = self.connector_config.bootstrap_server
+        port = self.connector_config.port
+
+        conf = {
+            "bootstrap.servers": f"{bootstrap}:{port}",
+            "client.id": socket.gethostname(),
+            "group.id": "your_group_id",
+            "enable.auto.commit": "false",
+            "auto.offset.reset": "earliest",
+            "message.max.bytes": 10485760,
+        }
+
+        if is_confluent:
+            kafka_api_key = self.connector_config.access_config.kafka_api_key
+            secret = self.connector_config.access_config.secret
+            conf["sasl.mechanism"] = "PLAIN"
+            conf["security.protocol"] = "SASL_SSL"
+            conf["sasl.username"] = kafka_api_key
+            conf["sasl.password"] = secret
+
+        consumer = Consumer(conf)
+        logger.debug(f"Kafka Consumer connected to bootstrap: {bootstrap}")
+        return consumer
+
+    @SourceConnectionError.wrap
+    def get_ingest_docs(self):
+        from confluent_kafka import KafkaError
+
+        consumer = self.kafka_consumer
+        running = True
+
+        collected = []
+        num_messages_to_consume = self.connector_config.num_messages_to_consume
+        logger.info(f"Config set for blocking on {num_messages_to_consume} messages")
+        # Consume specified number of messages
+        while running:
+            msg = consumer.poll(timeout=1.0)
```

Comment:
Should this timeout be set as a variable to be passed in?

Reply:
Added.